src/ipc/capnp/common-types.h

```diff
@@ -122,6 +122,36 @@ decltype(auto) CustomReadField(TypeList<UniValue>, Priority<1>, InvokeContext& i
         value.read(std::string_view{data.begin(), data.size()});
     });
 }
+
+//! Generic ::capnp::Data field builder for any class that a Span can be
+//! constructed from, particularly BaseHash and base_blob classes and
+//! subclasses. It's also used to serialize vector<char> set elements
+//! in GCSFilter::ElementSet and CBlockTemplate::vchCoinbaseCommitment.
+//!
+//! There is currently no corresponding ::capnp::Data CustomReadField function
+//! that works using Spans, because the bitcoin classes in the codebase like
+//! BaseHash and blob_blob that can converted /to/ Span don't currently have
+//! Span constructors that allow them to be constructed /from/ Span. If they
+//! did, it would simplify things. For example, a generic CustomReadField
+//! function could be written that would allow dropping specialized
+//! CustomReadField functions for types like PKHash.
+//!
+//! For the LocalType = vector<char> case, it's also not necessary to
+//! have ::capnp::Data CustomReadField function corresponding to this
```

Comment:
re: https://github.com/bitcoin/bitcoin/pull/30510#discussion_r1771916488

I rewrote the comment and tweaked implementation of this to avoid use of non-standard Span class, so this could be dropped and moved to libmultiprocess in the future. But leaving that for a followup since it will require other changes to libmultiprocess.